src/components/shared/buttons/OutlinedToggleButton.tsx

```diff
@@ -1,85 +1,49 @@
-    outlinedButtonBackground_disabled,
-    primaryColoredOutline,
-    primaryColoredOutline_disabled,
-    primaryColoredOutline_hovered,
 } from 'context/Theme';
-import { OutlinedToggleButtonProps } from './types';
 
-function OutlinedToggleButton({
-    children,
-    defaultStateSx,
-    disabled,
-    disabledStateSx,
-    selectedStateSx,
+const OutlinedToggleButton = styled(ToggleButton)(({
+    color,
     selected,
-    value,
-    onChange,
-    onClick,
-    ...props
-}: OutlinedToggleButtonProps) {
-    const theme = useTheme();
-
-    const defaultDisabledStateSx: SxProps<Theme> = selected
-        ? {
-              backgroundColor:
-                  outlinedButtonBackground_disabled[theme.palette.mode],
-              border: primaryColoredOutline_disabled[theme.palette.mode],
-              color: disabledButtonText_primary[theme.palette.mode],
-          }
-        : {
-              backgroundColor: 'none',
-              border: defaultOutline[theme.palette.mode],
-              color: disabledButtonText[theme.palette.mode],
-          };
+    theme,
+}) => {
+    const colorKey = color ?? 'primary';
 
-    let sx: SxProps<Theme> = {
-        px: '9px',
-        py: '3px',
-        border: intensifiedOutline[theme.palette.mode],
-        borderRadius: 2,
-        [`&.${toggleButtonClasses.selected}`]: selectedStateSx ?? {
-            'backgroundColor': outlinedButtonBackground[theme.palette.mode],
-            'border': primaryColoredOutline[theme.palette.mode],
-            'color': theme.palette.primary.main,
+    return {
+        'padding': '3px 9px',
+        'border': intensifiedOutline[theme.palette.mode],
+        'borderRadius': 4,
+        '&:hover': {
+            border: defaultOutline_hovered[theme.palette.mode],
+        },
+        [`&.${toggleButtonClasses.selected}`]: {
+            'backgroundColor': theme.palette[colorKey].alpha_12,
+            'border': `1px solid ${theme.palette[colorKey].alpha_50}`,
+            'color': theme.palette[colorKey].dark,
             '&:hover': {
-                border: primaryColoredOutline_hovered[theme.palette.mode],
+                border: `1px solid ${
+                    colorKey === 'success'
+                        ? theme.palette[colorKey].dark
+                        : theme.palette[colorKey].main
+                }`,
             },
         },
-        [`&.${toggleButtonClasses.disabled}`]:
-            disabledStateSx ?? defaultDisabledStateSx,
+        [`&.${toggleButtonClasses.disabled}`]: selected
+            ? {
+                  backgroundColor: theme.palette[colorKey].alpha_05,
+                  border: `1px solid ${theme.palette[colorKey].alpha_12}`,
+                  color: theme.palette[colorKey].alpha_26,
+              }
+            : {
+                  backgroundColor: 'none',
+                  border: defaultOutline[theme.palette.mode],
+                  color: disabledButtonText[theme.palette.mode],
+              },
```

Comment:
What is the case where `selected` is true but the class for selected is not applied? Hoping we could get the same styling with more static styling.

Reply:
The disabled styling is applied last and, thus, it overrides the selected styling.

Reply:
How? They have two different classes.